Add Arena export (via copy to clipboard)

KaelenProctor opened the pull request:
Implements #1181. For the mainboard of the cube it generates the text to copy/paste into Arena, and has a button to copy direct to browser clipboard.

As I would only classify myself as having a working man's knowledge of React/Hooks, happy to adjust in case there are more React-y ways of doing things.

### Testing

#### General

1) Use Sort/filter buttons apply to the Arena export when the modal opens. Tested several sort changes of the primary and secondary sorts, assuming the rest would work the same
2) Add and remove cards from the cube, the Arena export list is updated when modal opens
3) Multiple of the same card are exported and in Arena are counted together

#### Importing to Arena:
1) For each set on Arena (including extras like Jumpstart, Anthologies, and remasters) exported the card names via Scryfall. This got me approximately 170 cards per set. Tested over 40 sets.
2) Imported those names into a Cube
3) Exported to Arena via copy to clipboard
4) Imported to Arena

Issues encountered:
1) The backside of meld cards (ex. Mishra, Lost to Phyrexia) are not found on import to Arena. It does not have the backsides as "cards"
2) Cards ending with exclamation mark (ex. Fear, Fire, Foes!) cannot be imported (even using Arena export/import). Known bug open with Arena

### Screenshots

Export modal
![export](https://github.com/user-attachments/assets/aa94fc1f-6ffa-4d4f-83de-c82776681b18)

Modal after clicking copy to clipboard button
![export-after-copy](https://github.com/user-attachments/assets/47892e2c-678d-46ec-8b06-81cae632cfe9)

Example of filter applying when "Use filter" is enabled
![filter-applying](https://github.com/user-attachments/assets/b59851a9-00c1-45f3-86b8-b1c384864203)

Example of filter ignored when "Use filter" is off
![filter-not-applied](https://github.com/user-attachments/assets/7d6fda4d-0139-44ca-824a-ee50198082dd)

Sorting applied
![sorting](https://github.com/user-attachments/assets/2ac0f09a-3d61-4ede-b72b-5d5371bac279)

dekkerglen commented:
This is going to be obsoleted by tailwind css and ts migration, will hold off until that is done and then we can rebase

KaelenProctor commented:
@dekkerglen I've rebased the work against the master branch after the tailwindcss changes were merged in. Converted to Typescript while doing so, have to say the types are pretty nice to work with.
